onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -779,6 +779,42 @@ bool ReshapeOpSupportChecker::HasSupportedInputOutputsImpl(
 
 #pragma endregion
 
+#pragma region op_unsqueeze
+
+class UnsqueezeOpSupportChecker : public BaseOpSupportChecker {
+ private:
+  bool IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                         const OpSupportCheckParams& params) const override;
+};
+
+bool UnsqueezeOpSupportChecker::IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                                                  const OpSupportCheckParams& /* params */) const {
+  const auto& inputs = node_unit.Inputs();
+  Shape input_shape;
+  if (!GetShape(inputs[0].node_arg, input_shape))
+    return false;
+
+  const auto input_rank = input_shape.size();
+  if (input_rank > 4 || input_rank == 0) {
+    LOGS_DEFAULT(VERBOSE) << "Unsqueeze only supports 1-4d shape, input is "
+                          << input_rank << "d shape";
+    return false;
+  }
+
+  // Unsqueeze opset 13 uses input 1 as axes, if we have input 1 then it needs to be an initializer
+  if (node_unit.SinceVersion() > 12 && inputs.size() > 1) {
```

Comment:
FYI: Ed noticed that this check doesn't include making sure it's a constant initializer. We will clean that up in a separate PR though.